### ISSUE TITLE
[question] conan 2 output to file

### ISSUE THREAD

**Issue by araffin** [conan-io/conan#16098]
### What is your question?

Hello,

Could you point me to the discussion/reasoning behind the fact that conan v2 doesn't offer the option to output format to files, but rather outputs things to stdout (and the rest to stderr)?

From a user point of view, I already encountered issues that https://github.com/conan-io/conan/pull/15704 tries to solve ( https://github.com/conan-io/conan/issues/15671) but the current solution (overriding python built-in) looks more like a hack to me that can break in other cases, for instance if the user directly output things to stdout.

In short, I'm curious about the reasons behind this design choice and if output to a file is a planned feature?

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
> Could you point me to the discussion/reasoning behind the fact that conan v2 doesn't offer the option to output format to files, but rather outputs things to stdout (and the rest to stderr)?

To follow more standard command line tools approaches, to be able to redirect output instead of generating files. It is something that can be done in CI or other automation, to avoid the creation of temporary files in disk, just redirect to memory and parse and use the information directly from the output stream in memory, not needing to read from disk (that can pollute the checkout for example if you don't recall to remove it later).

> In short, I'm curious about the reasons behind this design choice and if output to a file is a planned feature?

No, it is not planned at the moment, the standard ``cmd > file`` redirection has been working well so far. The ``print()`` issues were just small unexpected things, because users doing undocumented things instead of using the documented ``self.output``.

**Comment by araffin** [comment]
thanks for the swift answer =).

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/17507 adds a new ``--output-file`` to enforce writing the result in that file irrespective of the ``stdout`` redirections or other possible issues. It will be in next Conan 2.12. Thanks for the feedback!
